runtime/embed/process.go

```diff
@@ -19,7 +21,8 @@ func process(worker *Worker, loggingBackend sqlq.LogBackend) (shutdown func(time
 	//
 	// quit is used to unblock the primary goroutine waiting on semaphore
 	// abort is used to terminate any handler routine that has not completed even after timeout for shutdown has expired
-	var quit, abort = make(chan struct{}), make(chan struct{})
+	// cancelled is used to terminate the job routine that have a status equal to cancelling
+	var quit, abort, cancelled = make(chan struct{}), make(chan struct{}), make(chan struct{})
```

Comment:
shouldn't the `cancelled` channel be created per-job? that ways we `close()` this channel only if that `job` is marked as `cancelling` and it wouldn't affect other running jobs.